File: src/main/java/org/wcardinal/controller/ComponentFactory.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller;

/**
 * A factory that creates/destroys the specified class T as a component dynamically.
 *
 * @param <T> component class to be created/destroyed dynamically
 */
public interface ComponentFactory<T> extends Factory<T> {

}

File: src/main/typescript/wcardinal/controller/component-factory.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { Component } from "./component";
import { Factory } from "./factory";

/**
 * A component factory that creates/destroys components dynamically.
 */
export interface ComponentFactory extends Factory<Component> {}
